tests/samplers_tests/test_samplers.py

```diff
@@ -1010,14 +1085,23 @@ def objective(trial: Trial) -> float:
         return a + b + c + d + e + f + g
 
     hash_dict[k] = hash("nondeterministic hash")
-    sampler = sampler_class_with_seed[sampler_class_index](1)
+    sampler = cast(
+        Callable[[int], BaseSampler],
+        [
+            sampler_class
+            for sampler_class in sampler_class_with_seed
+            if "[" + cast(str, sampler_class.id) + "]" in test_name
+        ][0].values[0],
+    )(1)
     study = optuna.create_study(sampler=sampler)
     study.optimize(objective, n_trials=20)
     sequence_dict[k] = list(study.trials[-1].params.values())
 
 
-@pytest.mark.parametrize("sampler_class_index", range(len(sampler_class_with_seed)))
-def test_reproducible_in_other_process(sampler_class_index: int) -> None:
+@parametrize_sampler_with_seed
+def test_reproducible_in_other_process(
+    request: FixtureRequest, sampler_class: Callable[[int], BaseSampler]
```

Comment:
Why do you fix this test as now? Could you explain a bit?

Reply:
We need to directly use `parametrize_sampler_with_seed` to deselect the integration tests. BTW, this change seems complicated. I will refine the implementation.